visualizer/semanticsHelper.js

```diff
@@ -259,12 +289,82 @@
     return pexpr;
   }
 
+  // Build the action that called from the wrapper, which includes renaming the
+  // operation arguments, and actual action body that user typed.
+  function buildAction(opArgs, enclosedActionArgStr, actionBody) {
+    // Constructs the code that renames the operation arguments, so we could refer
+    // them by name directly. i.e. `var <argument name> = this.args.<argument name>`
+    var renameOpArgStr = opArgs &&  Object.keys(opArgs).map(function(argName) {
+      return 'var ' + argName + ' = this.args.' + argName + ';';
+    }).join('\n');
+
+    var actionStr = 'function' + enclosedActionArgStr + '{\n' +
+        renameOpArgStr + '\n' + actionBody +
+      '\n}';
+    return eval('(' + actionStr + ')');    // eslint-disable-line no-eval
```

Comment:
I think you should actually use the Function constructor here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function. It's basically like `eval`, but it's _slightly_ safer, because the code doesn't have access to local scope variables (though it still has access to the global object).